openedx/core/lib/api_utils.py

```diff
@@ -0,0 +1,73 @@
+"""Helper functions to get data from APIs"""
+from __future__ import unicode_literals
+import logging
+
+from django.core.cache import cache
+from edx_rest_api_client.client import EdxRestApiClient
+
+from openedx.core.lib.token_utils import get_id_token
+
+
+log = logging.getLogger(__name__)
+
+
+def get_api_data(api_config, user, api_name, resource, querystring=None, use_cache=False, cache_key=None):
+    """Fetch the data from the API using provided API Configuration and
+    resource.
+
+    Arguments:
+        api_config: The configuration which will be user for requesting data.
+        user (User): The user to authenticate as when requesting data.
+        api_name: Name fo the api to be use for logging.
+        resource: API resource to from where data will be requested.
+        querystring: Querystring parameters that might be required to request
+            data.
+        use_cache: Will be used to decide whether to cache the response data
+            or not.
+        cache_key: cache key to be used to save response data.
+
+    Returns:
+        list of dict, representing data returned by the API.
+    """
+    no_data = []
+
+    if not api_config.enabled:
+        log.warning('%s configuration is disabled.', api_name)
+        return no_data
+
+    if use_cache:
+        if cache_key:
+            cached = cache.get(api_config.CACHE_KEY)
+            if cached is not None:
+                return cached
+        else:
+            log.warning('Cache key does not provided to cache data.')
```

Comment:
This message needs to be revised, but I think you'll be removing it when making the changes I'm describing.